### PR TITLE
REF: avoid _with_infer constructor

### PR DESCRIPTION
**Pull request by jbrockmendel**
Trying to get down to Just One constructor.

### CODE REVIEW

File: pandas/tests/io/test_stata.py [hunk 1]
```diff
@@ -1144,7 +1143,7 @@ def _convert_categorical(from_frame: DataFrame) -> DataFrame:
             if is_categorical_dtype(ser.dtype):
                 cat = ser._values.remove_unused_categories()
                 if cat.categories.dtype == object:
-                    categories = ensure_index(cat.categories._values)
+                    categories = pd.Index._with_infer(cat.categories._values)
```
Comment (WillAyd) [inline pandas/tests/io/test_stata.py:1146]: Is this supposed to have _with_infer still?

**Review by WillAyd — COMMENTED** [review]
looks good

File: pandas/tests/io/test_stata.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/io/test_stata.py:1146]: yes

File: pandas/core/indexes/base.py [hunk 2]
```diff
@@ -6909,9 +6910,10 @@ def ensure_index(index_like: Axes, copy: bool = False) -> Index:
 
             return MultiIndex.from_arrays(index_like)
         else:
-            return Index._with_infer(index_like, copy=copy, tupleize_cols=False)
+            return Index(index_like, copy=copy, tupleize_cols=False)
     else:
-        return Index._with_infer(index_like, copy=copy)
+        # with_infer needed for stata tests
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/core/indexes/base.py:6915]: good catch, will update

[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/core/indexes/base.py]: updated+green

**Review by mroeschke — APPROVED** [review]
LGTM merge when ready @WillAyd

**Comment by WillAyd** [comment]
Thanks @jbrockmendel

**Pull request merged**